packages/agents-core/src/run.ts

```diff
@@ -127,7 +127,7 @@ export type RunConfig = {
   /**
    * An optional dictionary of additional metadata to include with the trace.
    */
-  traceMetadata?: Record<string, any>;
+  traceMetadata?: Record<string, string>;
```

Comment:
Confirmed the server side behavior and found that only string values are allowed here. Also, it does not support nested data.